README.md
# Korus-UI

Project's repository: [GitHub][15]

Korus-UI is a customizable UI component library with built-in forms handling and some other niceties.

Supports TypeScript@^3.8.3 and React@^16.8.0 (the one with hooks).

![](https://github.com/kksng/leda/workflows/.github/workflows/ci.yml/badge.svg)

## Key Features

- Full customization of library components for your project in [3 different ways][6]
- Concise and functional forms with validation (significantly less code than in popular solutions, with more flexibility and functionality)
- Improved layout (less code, easier to work with CSS classes)
- Uniform API (if you worked with one component you know how to work with the rest)
- About 70 components for building forms and layout

### Everything that you wanted from validation you can find in Korus-UI

- Simplicity
- Field validation by function or RegExp
- There are ready-made validators
- One or more validators for each field with its own messages
- Custom messages (text and styles)
- Setting the field validity from the outside using the isValid attribute
- Validation of components in the unmounted state
- Validation of fields on loss of focus, form submission, or any custom event in the app
- Scroll to invalid fields when submitting a form
- One-click validation of multiple forms
- Helper functions for validation of passed values


## Technologies

- React@^16.8.0
- TypeScript@^3.8.3


## Documentation

The Korus-UI docs are located at [https://opensource.esphere.ru/korus-ui/][1].

All available features are documented in Russian. English version is on it's way.

- [Introduction][7]
- [API Reference][8]
- [CSS classes][9]
- [Forms][10]
- [Validation][11]
- [Refs and DOM][12]


## Code Style

We use [Airbnb JavaScript Style Guide][16]


## Installation

```
npm install korus-ui
```
Use the `L` namespace to distinguish library components from other components in your application.

```
import * as L from 'korus-ui';
```

## Development mode

- Clone or download the repo
- Run `npm i` to install dependencies
- Run `npm start` to launch dev-server with demo examples on `http://localhost:9000/`


### Available NPM scripts

| script | description |
|----------------|----------------|
| npm start | runs project in development mode |
| npm run build | builds project |
| npm run check | checks eslint and typescript errors, runs jest tests |
| npm run cypress | runs cypress tests |
| npm run coverage | shows coverage after running cypress tests |


## File Structure

```
├── demo                         : demo examples
│   ├── components                   : demo examples for dev mode
│   │   ├── A                           : demo examples of A component
│   │   ├── AutoComplete                : demo examples of AutoComplete component
│   │   ├── Button                      : demo examples of Button component
│   │   ├── ButtonGroup                 : demo examples of ButtonGroup component
│   │   ├── CheckBox                    : demo examples of CheckBox component
│   │   ├── Collapse                    : demo examples of Collapse component
│   │   ├── Collapsible                 : demo examples of Collapsible component
│   │   ├── Currency                    : demo examples of Currency component
│   │   ├── DatePicker                  : demo examples of DatePicker component
│   │   ├── DateRange                   : demo examples of DateRange component
│   │   ├── DateTimePicker              : demo examples of DateTimePicker component
│   │   ├── DateTimeRange               : demo examples of DateTimeRange component
│   │   ├── DropDown                    : demo examples of DropDown component
│   │   ├── DropDownLink                : demo examples of DropDownLink component
│   │   ├── DropDownSelect              : demo examples of DropDownSelect component
│   │   ├── DropZone                    : demo examples of DropZone component
│   │   ├── FileDrop                    : demo examples of FileDrop component
│   │   ├── FileUpload                  : demo examples of FileUpload component
│   │   ├── FileList                    : demo examples of FileList component
│   │   ├── HTMLTags                    : demo examples of HTMLTags component
│   │   ├── Input                       : demo examples of Input component
│   │   ├── LedaProvider                : demo examples of LedaProvider component
│   │   ├── Loader                      : demo examples of Loader component
│   │   ├── MaskedInput                 : demo examples of MaskedInput component
│   │   ├── Modal                       : demo examples of Modal component
│   │   ├── MultiSelect                 : demo examples of MultiSelect component
│   │   ├── Notifications               : demo examples of Notifications component
│   │   ├── NumericRange                : demo examples of NumericRange component
│   │   ├── NumericTextBox              : demo examples of NumericTextBox component
│   │   ├── Pagination                  : demo examples of Pagination component
│   │   ├── ProgressBar                 : demo examples of ProgressBar component
│   │   ├── RadioGroup                  : demo examples of RadioGroup component
│   │   ├── Rating                      : demo examples of Rating component
│   │   ├── Slider                      : demo examples of Slider component
│   │   ├── StatusBar                   : demo examples of StatusBar component
│   │   ├── StickyPanel                 : demo examples of StickyPanel component
│   │   ├── Svg                         : demo examples of Svg component
│   │   ├── Switcher                    : demo examples of Switcher component
│   │   ├── Table                       : demo examples of Table component
│   │   ├── Tabs                        : demo examples of Tabs component
│   │   ├── Tags                        : demo examples of Tags component
│   │   ├── Textarea                    : demo examples of Textarea component
│   │   ├── TimePicker                  : demo examples of TimePicker component
│   │   ├── TimeRange                   : demo examples of TimeRange component
│   │   ├── Tooltip                     : demo examples of Tooltip component
│   │   ├── Tour                        : demo examples of Tour component
│   │   ├── Validation                  : demo examples of Validation component
│   │   ├── VStepper                    : demo examples of VStepper component
│   │   ├── Wizard                      : demo examples of Wizard component
│   │   ├── Div.tsx                     : demo examples of Div component
│   │   ├── Dl.tsx                      : demo examples of Dl component
│   │   ├── index.ts                    : export of demo examples
│   │   ├── Navigation.tsx              : demo examples of Navigation component
│   │   ├── StateButtonGroup.tsx        : helper component for demo examples
│   │   ├── Story.tsx                   : layout for demo examples
│   │   └── Styles.tsx                  : styles for demo examples
│   │
│   ├── cypress                      : demo examples for cypress tests
│   │   ├── AutoComplete                : demo examples of AutoComplete component for cypress tests
│   │   ├── Button                      : demo examples of Button component for cypress tests
│   │   ├── ButtonGroup                 : demo examples of ButtonGroup component for cypress tests
│   │   ├── Checkbox                    : demo examples of Checkbox component for cypress tests
│   │   ├── Collapse                    : demo examples of Collapse component for cypress tests
│   │   ├── Currency                    : demo examples of Currency component for cypress tests
│   │   ├── DatePicker                  : demo examples of DatePicker component for cypress tests
│   │   ├── DateRange                   : demo examples of DateRange component for cypress tests
│   │   ├── DateTimePicker              : demo examples of DateTimePicker component for cypress tests

CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [0.46.0] - 2021-07-07

### Added
- MultiSelect: Added header and footer in checkboxes list
- MultiSelect: Refactoring
- FileList: Added new component
- Storybook: Added documentation for FileList component
- DropZone: Add new styles
- DropDown: Add new styles
- Password: Add new styles
- Pagination: Add new styles

## [0.45.1] - 2021-04-29

### Fixed
- MaskedInput: Fixed autocomplete


## [0.45.0] - 2021-04-26

### Added
- CheckBoxTree: Added new component


## [0.44.0] - 2021-04-12

### Added
- DropDown: Added class disabled

### Fixed
- NumericTextBox: Fixed rounding numbers on change event


## [0.43.0] - 2021-03-29

### Added
- LinkTree: Added new component
- Drawer: Added new component

### Fixed
- DropDownSelect: Fixed long strings display


## [0.42.0] - 2021-03-15

### Added
- CountDown: Added new component

### Fixed
- Tabs: Fixed scroll


## [0.41.0] - 2021-02-25

### Added
- Menu: Added new component
- Menu: Close dropdown items on scroll

### Changed
- Modal: Prop 'size' adds class to div.modal-window instead of style attribute

### Fixed
- DatePicker: Fixed date selection from previous month


## [0.40.0] - 2021-02-09

### Added
- Validation: added scroll to invalid DropZone and FileDrop fields
- FileUpload: Added maxFileNameLength prop
- MaskedInput: Added clear button

### Changed
- README: Updated browsers list in README.md
- Updated .npmignore

### Fixed
- MaskedInput: Fixed backspace
- Fixed test coverage report for Windows
- Tabs: Fixed scroll
- MaskedInput: Remove mask on reset


## [0.39.0] - 2021-01-26

### Added
- TimeRange, TimePicker: Added autocompletion with 0


## [0.38.0] - 2021-01-12

### Changed
- Refactoring: Renamed leda folder to korus-ui

### Fixed
- Tour: Fixed scroll behaviour when active element is inside modal window
- TimeRange: Fixed behaviour on Tab click
- DateRange: Fixed time for date values
- TimeRange, TimePicker: Added max limits for hours, minutes seconds
- Changed file structure in README


## [0.37.0] - 2020-12-21

### Added
- Tour: Added step changing animation. Block page while step loading
- VStepper: Added type warning, new props footerContent and nextStepType to VStepperItem
- Refactoring: Added changelog generator

### Changed
- Storybook: Changed library name to Korus-ui

### Fixed
- ToolTip: Fixed tooltip display
- Tabs: Fixed tabs display and scroll in IE
- Modal, DemoPage: validate DemoPage Modal name field
- FileDrop: Prevented firing change event before file upload
- FileDrop: Fixed reset handler
- Storybook: Fixed bug that prevented assets folder from being copied to build
- Tour: Fixed IE component behavior.
- NumericTextBox, MaskedInput: Fixed demo examples, removed unused props prefixRender and suffixRender


## [0.36.0] - 2020-12-07

### Added
- Added release policy to README
- Completed description in README.md: technologies, git workflow, project's structure

### Changed
- CONTRIBUTING.md, Issue template: Update GitHub issue template

### Fixed
- Input: Fixed paste event handler for Input with maxLength prop
- MaskedInput: Fixed clearing incomplete mask value by setting null value or calling reset function
- DropDown: Fixed IE11 bug. Dropdown list should be visible.
- MultiSelect: Fixed styles for IE
- Tooltip: Fixed displaying in IE
- MultiSelect: Fixed working with object data
- MaskedInput: Fixed cursor position


## [0.35.0] - 2020-11-23

### Added
- Tour: Added stepDelay prop, added positions: top-left, bottom-left, top-center, bottom-center
- DropZone: Added isLoading, loadingProgress and loadingViewRender props

### Fixed
- Validation: Fixed displaying of invalidMessage with isValid prop set to false
- NumericRange: Fixed null value display bug

snapshots.js
module.exports = {
  "__version": "5.2.0",
  "AutoComplete": {
    "Display": {
      "Should render placeholder": {
        "1": "<input aria-invalid=\"false\"\n  aria-required=\"true\"\n  autocomplete=\"off\"\n  class=\"autocomplete-input\"\n  form=\"myForm\"\n  name=\"AutoComplete2\"\n  placeholder=\"Type your city...\"\n  value=\"\">"
      },
      "Should render ClearButton": {
        "1": "<i class=\"autocomplete-clear-icon\"></i>"
      },
      "Should render SuggestionList when isOpen": {
        "1": "<ul class=\"suggestion-list\">\n  <li class=\"suggestion-item\">Tokyo</li>\n  <li class=\"suggestion-item\">Delhi</li>\n</ul>"
      },
      "Should render loader when isLoading": {
        "1": "<div class=\"loader-container\"><span class=\"loader-element\"></span></div>"
      },
      "noSuggestionsRender": {
        "Should render defaultMessage": {
          "1": "<div class=\"suggestion-wrapper visible\">\n  <div class=\"nodata\">Ничего не найдено</div>\n</div>"
        },
        "Should render customMessage": {
          "1": "<div class=\"suggestion-wrapper visible\">\n  <div class=\"txt-center inner nodata\">набери что-то, что я знаю</div>\n</div>"
        }
      }
    }
  },
  "Button": {
    "Display": {
      "Should render button": {
        "1": "<button type=\"button\"\n  id=\"basicButton\"\n  class=\"button-wrapper\">Click!</button>"
      },
      "Should render button when isLoading": {
        "1": "<button type=\"button\"\n  id=\"loadButton\"\n  class=\"button-wrapper loading\">isLoading</button>"
      },
      "Should render button when isDisabled": {
        "1": "<button type=\"button\"\n  id=\"disabledButton\"\n  class=\"button-wrapper disabled\">isDisabled</button>"
      },
      "Should render with form": {
        "1": "<div class=\"input-wrapper\">\n  <div class=\"input-element-wrapper required\"><input placeholder=\"form1 Input1\"\n      aria-invalid=\"false\"\n      aria-required=\"true\"\n      class=\"input-element\"\n      form=\"form1\"\n      name=\"Input1\"\n      value=\"\"></div>\n</div>"
      }
    }
  },
  "ButtonGroup": {
    "Display": {
      "Should render all the buttons in the group": {
        "1": "<button type=\"button\"\n  class=\"button-wrapper button-group-item first last\">one</button>"
      }
    }
  },
  "CheckBox": {
    "Display": {
      "Should render elements inside the checkbox": {
        "1": "<button type=\"button\"\n  class=\"button-wrapper loading\">isLoading</button>"
      },
      "Should render semi": {
        "1": "<label for=\"checkBoxSemi\"\n  class=\"checkbox-label semi\">isSemi</label>"
      },
      "Should render disabled checkbox": {
        "1": "disabled"
      }
    }
  },
  "DatePicker": {
    "Display": {
      "Should render datepicker": {
        "1": "<input class=\"datepicker-input\"\n  maxlength=\"11\"\n  placeholder=\"Type your date...\"\n  aria-invalid=\"false\"\n  name=\"firstDatePicker\"\n  value=\"\">"
      },
      "Should render placeholder": {
        "1": "<input class=\"datepicker-input\"\n  maxlength=\"11\"\n  placeholder=\"Type your date...\"\n  aria-invalid=\"false\"\n  name=\"firstDatePicker\"\n  value=\"\">"
      }
    },
    "Interaction": {
      "Should work with different date formats": {
        "1": "<input class=\"datepicker-input\"\n  maxlength=\"11\"\n  aria-invalid=\"false\"\n  name=\"secondDatePicker\"\n  value=\"11.11.1111\">",
        "2": "<input class=\"datepicker-input\"\n  maxlength=\"39\"\n  aria-invalid=\"false\"\n  name=\"openedCalendar\"\n  value=\"11-е число  11-го месяца  1111-го года\">"
      }
    }
  },
  "DateTimeRange": {
    "Display": {
      "Should render DateTimeRange": {
        "1": "<div class=\"datepicker-wrapper\">\n  <div class=\"datepicker-input-wrapper\"><input class=\"datepicker-input\"\n      maxlength=\"17\"\n      aria-invalid=\"false\"\n      aria-required=\"true\"\n      name=\"DateTimeRange-to\"\n      value=\"15.05.2018 16:30\"><span class=\"datepicker-icons-wrapper\"><span\n        class=\"datepicker-calendar-icon\"></span></span></div>\n</div>"
      }
    }
  },
  "DateTimePicker": {
    "Display": {
      "Should render DateTimePicker": {
        "1": "<input class=\"datepicker-input\"\n  maxlength=\"18\"\n  placeholder=\"datetimepicker\"\n  data-test=\"dp1\"\n  form=\"date-form\"\n  aria-invalid=\"false\"\n  aria-required=\"true\"\n  name=\"datetimepicker\"\n  value=\"23.10.18 14:30:25\">"
      }
    }
  },
  "DateRange": {
    "Display": {
      "Should render DateRange": {
        "1": "<div class=\"datepicker-wrapper\">\n  <div class=\"datepicker-input-wrapper\"><input class=\"datepicker-input\"\n      maxlength=\"11\"\n      placeholder=\"Type your date...\"\n      aria-invalid=\"false\"\n      name=\"MinMaxDatePicker-to\"\n      value=\"\"><span class=\"datepicker-icons-wrapper\"><span\n        class=\"datepicker-calendar-icon\"></span></span></div>\n  <div class=\"calendar-wrapper visible\">\n    <div class=\"calendar-nav\"><span class=\"calendar-prev-button\"><i\n          class=\"calendar-prev-icon\"></i></span><span class=\"calendar-title\">Май\n        2012</span><span class=\"calendar-next-button disabled-button\"><i\n          class=\"calendar-next-icon\"></i></span></div>\n    <div class=\"calendar-week-days\">\n      <div title=\"Понедельник\"\n        class=\"calendar-date-cell\">Пн</div>\n      <div title=\"Вторник\"\n        class=\"calendar-date-cell\">Вт</div>\n      <div title=\"Среда\"\n        class=\"calendar-date-cell\">Ср</div>\n      <div title=\"Четверг\"\n        class=\"calendar-date-cell\">Чт</div>\n      <div title=\"Пятница\"\n        class=\"calendar-date-cell\">Пт</div>\n      <div title=\"Суббота\"\n        class=\"calendar-date-cell\">Сб</div>\n      <div title=\"Воскресенье\"\n        class=\"calendar-date-cell\">Вс</div>\n    </div>\n    <div class=\"calendar-month-dates\">\n      <div class=\"calendar-dates-row\">\n        <div class=\"calendar-date-cell different-month-date\"\n          title=\"30 апреля 2012\">30</div>\n        <div class=\"calendar-date-cell\"\n          title=\"1 мая 2012\">1</div>\n        <div class=\"calendar-date-cell\"\n          title=\"2 мая 2012\">2</div>\n        <div class=\"calendar-date-cell\"\n          title=\"3 мая 2012\">3</div>\n        <div class=\"calendar-date-cell active\"\n          title=\"4 мая 2012\">4</div>\n        <div class=\"calendar-date-cell disabled-date\">5</div>\n        <div class=\"calendar-date-cell disabled-date\">6</div>\n      </div>\n      <div class=\"calendar-dates-row\">\n        <div class=\"calendar-date-cell disabled-date\">7</div>\n        <div class=\"calendar-date-cell disabled-date\">8</div>\n        <div class=\"calendar-date-cell disabled-date\">9</div>\n        <div class=\"calendar-date-cell disabled-date\">10</div>\n        <div class=\"calendar-date-cell disabled-date\">11</div>\n        <div class=\"calendar-date-cell disabled-date\">12</div>\n        <div class=\"calendar-date-cell disabled-date\">13</div>\n      </div>\n      <div class=\"calendar-dates-row\">\n        <div class=\"calendar-date-cell disabled-date\">14</div>\n        <div class=\"calendar-date-cell disabled-date\">15</div>\n        <div class=\"calendar-date-cell disabled-date\">16</div>\n        <div class=\"calendar-date-cell disabled-date\">17</div>\n        <div class=\"calendar-date-cell disabled-date\">18</div>\n        <div class=\"calendar-date-cell disabled-date\">19</div>\n        <div class=\"calendar-date-cell disabled-date\">20</div>\n      </div>\n      <div class=\"calendar-dates-row\">\n        <div class=\"calendar-date-cell disabled-date\">21</div>\n        <div class=\"calendar-date-cell disabled-date\">22</div>\n        <div class=\"calendar-date-cell disabled-date\">23</div>\n        <div class=\"calendar-date-cell disabled-date\">24</div>\n        <div class=\"calendar-date-cell disabled-date\">25</div>\n        <div class=\"calendar-date-cell disabled-date\">26</div>\n        <div class=\"calendar-date-cell disabled-date\">27</div>\n      </div>\n      <div class=\"calendar-dates-row\">\n        <div class=\"calendar-date-cell disabled-date\">28</div>\n        <div class=\"calendar-date-cell disabled-date\">29</div>\n        <div class=\"calendar-date-cell disabled-date\">30</div>\n        <div class=\"calendar-date-cell disabled-date\">31</div>\n        <div class=\"calendar-date-cell disabled-date\">1</div>\n        <div class=\"calendar-date-cell disabled-date\">2</div>\n        <div class=\"calendar-date-cell disabled-date\">3</div>\n      </div>\n    </div>\n  </div>\n</div>"
      },
      "Should show different required messages": {
        "1": "<div class=\"datepicker-wrapper\">\n  <div class=\"datepicker-input-wrapper danger required\"><input\n      class=\"datepicker-input\"\n      maxlength=\"11\"\n      placeholder=\"ThirdDateRange\"\n      form=\"112\"\n      aria-invalid=\"true\"\n      aria-required=\"true\"\n      name=\"ThirdDateRange-to\"\n      value=\"\"><span class=\"datepicker-icons-wrapper\"><span\n        class=\"datepicker-calendar-icon\"></span></span></div><span\n    class=\"invalid-message-list\"><span class=\"invalid-message-item\">second\n      message</span></span>\n</div>"
      },
      "Should show required message when isRequired": {
        "1": "<div class=\"datepicker-wrapper\">\n  <div class=\"datepicker-input-wrapper danger required\"><input\n      class=\"datepicker-input\"\n      maxlength=\"11\"\n      placeholder=\"Type something...\"\n      form=\"111\"\n      aria-invalid=\"true\"\n      aria-required=\"true\"\n      name=\"secondDatePicker\"\n      value=\"\"><span class=\"datepicker-icons-wrapper\"><span\n        class=\"datepicker-calendar-icon\"></span></span></div><span\n    class=\"invalid-message-list\"><span class=\"invalid-message-item\">required\n      message</span></span>\n</div>"
      }
    }
  },
  "DropDownSelect": {
    "Display": {
      "Should render the component": {
        "1": "<div class=\"dropdownselect-input-wrapper\"><input id=\"DDSBoundingContainerRef\"\n    aria-invalid=\"false\"\n    aria-required=\"false\"\n    autocomplete=\"off\"\n    class=\"dropdownselect-input\"\n    value=\"\"><span class=\"dropdownselect-select-icon closed\"></span></div>"
      },
      "Should render value": {
        "1": "<input id=\"DDSCompareObjectsBy\"\n  aria-invalid=\"false\"\n  aria-required=\"false\"\n  autocomplete=\"off\"\n  class=\"dropdownselect-input\"\n  placeholder=\"\"\n  readonly=\"\"\n  value=\"London\">"
      },
      "Should render data": {
        "1": "<div class=\"suggestion-wrapper visible\">\n  <ul class=\"suggestion-list\">\n    <li class=\"suggestion-item\">Moscow</li>\n    <li class=\"suggestion-item\">Minsk</li>\n    <li class=\"suggestion-item highlighted selected\">London</li>\n    <li class=\"suggestion-item\">Berlin</li>\n    <li class=\"suggestion-item\">Paris</li>\n    <li class=\"suggestion-item\">Stockholm</li>\n    <li class=\"suggestion-item\">Madrid</li>\n    <li class=\"suggestion-item\">Madrid</li>\n  </ul>\n</div>"
      }
    },
    "Should render loader when isLoading": {
      "1": "<div class=\"loader-container\"><span class=\"loader-element\"></span></div>"
    }
  },
  "DropZone": {
    "Display": {
      "DropZone should be displayed": {
        "1": "<div class=\"dropzone-wrapper required\">\n  <div tabindex=\"0\"\n    class=\"dropzone-file-upload\"><input form=\"dropzone-form\"\n      name=\"uncontrolledWithValidation\"\n      multiple=\"\"\n      type=\"file\"\n      autocomplete=\"off\"\n      tabindex=\"-1\"\n      style=\"display: none;\"><span class=\"dropzone-icon\"></span>\n    <div><span>Перетащите файлы или</span><a>&nbsp;нажмите здесь</a>\n      <div style=\"display: none;\"></div><i class=\"dropzone-hint\"></i>\n    </div>\n  </div>\n</div>"
      },
      "Should render file list": {
        "1": "<ul class=\"rejected-list margin-none\">\n  <li><span class=\"dropzone-rejected-message txt-danger\">'rejected file': -\n      Неизвестная ошибка</span></li>\n</ul>",
        "2": "<ul>\n  <li>\n    <div style=\"display: none;\"></div><a\n      class=\"pointer dropzone-delete-icon-wrapper\"><i\n        class=\"dropzone-delete-icon\"></i></a>\n    <div style=\"display: none;\"></div><a theme=\"dropzone-file-download internal\"\n      download=\"external file\"\n      href=\"external file link\">external file</a>\n  </li>\n</ul>"
      },
      "Should render customize upload button": {
        "1": "<button type=\"button\"\n  class=\"button-wrapper customized-button\">Drop Me</button>"
      },
      "Should render customize description text": {
        "1": "<span class=\"customized-dropzone\">Drop here</span>"
      }
    }
  },
  "FileUpload": {
    "Display": {
      "Should render component in controlled mode": {
        "1": "<a class=\"controlled\"\n  aria-invalid=\"false\"><span>Загрузить</span>\n  <div tabindex=\"0\"><input id=\"controlledFileUpload\"\n      accept=\".jpg, .gif, .png, .txt, .jpeg\"\n      type=\"file\"\n      autocomplete=\"off\"\n      tabindex=\"-1\"\n      style=\"display: none;\"></div>\n</a>"
      },
      "Should render customizated component": {
        "1": "<button type=\"button\"\n  aria-invalid=\"false\"\n  class=\"button-wrapper warning custom\"><span>Загрузить?</span>\n  <div tabindex=\"0\"><input id=\"customFileUpload\"\n      accept=\".jpg, .gif, .png\"\n      type=\"file\"\n      autocomplete=\"off\"\n      tabindex=\"-1\"\n      style=\"display: none;\"></div>\n</button>"
      },
      "Should render loading when using isLoading prop": {
        "1": "<a class=\"loaded underlining className\"\n  aria-invalid=\"false\"><span>Идёт загрузка...</span>\n  <div tabindex=\"0\"><input type=\"file\"\n      autocomplete=\"off\"\n      tabindex=\"-1\"\n      style=\"display: none;\"></div>\n</a>"
      }
    }
  },
  "Input": {
    "Display": {
      "Input should be displayed": {
        "1": "<div class=\"input-wrapper\">\n  <div class=\"input-element-wrapper\"><input id=\"only-numbers\"\n      placeholder=\"only numbers\"\n      aria-invalid=\"false\"\n      class=\"input-element\"\n      value=\"\"></div>\n</div>"
      },
      "isDisabled must set the component to disabled (the component is visible on the form, but it is inactive, there is no way to enter text)": {
        "1": "<div class=\"input-element-wrapper disabled\"><input id=\"isDisabled\"\n    placeholder=\"isDisabled\"\n    aria-invalid=\"false\"\n    class=\"input-element\"\n    disabled=\"\"\n    value=\"\"></div>"
      },
      "Should render component in controlled mode": {
        "1": "<div class=\"input-element-wrapper\"><input id=\"controlledMode\"\n    aria-invalid=\"false\"\n    class=\"input-element\"\n    value=\"Controlled value\"></div>"
      }
    }
  },
  "MaskedInput": {
    "Display": {
      "Should render component without placeholder": {
        "1": "<div class=\"masked-input-wrapper focused\"><input class=\"masked-input-element\"\n    maxlength=\"20\"\n    id=\"MICardNumberControlled\"\n    aria-invalid=\"false\"\n    value=\"____-____-____-____\"></div>"
      },
      "Should render placeholder": {
        "1": "<div class=\"masked-wrapper\">\n  <div class=\"masked-input-wrapper\"><input class=\"masked-input-element\"\n      maxlength=\"15\"\n      placeholder=\"___-___-___ __\"\n      id=\"MIControlledInsurance\"\n      aria-invalid=\"false\"\n      value=\"\"></div>\n</div>"
      },
      "Should render disabled state": {
        "1": "<div class=\"masked-input-wrapper disabled required\"><input\n    class=\"masked-input-element\"\n    maxlength=\"19\"\n    placeholder=\"+7 (___)-___-__-__\"\n    form=\"my-form\"\n    aria-invalid=\"false\"\n    aria-required=\"true\"\n    name=\"MINotControlledPhone\"\n    value=\"\"\n    disabled=\"\"></div>"
      }
    }
  },
  "Slider": {
    "Display": {
      "Should render the component": {
        "1": "<div class=\"slider-container\"\n  style=\"position: relative;\">\n  <div class=\"slider-track active\"\n    style=\"position: absolute; left: 0px; right: 385.5px;\"></div>\n  <div class=\"slider-track\"\n    style=\"position: absolute; left: 385.5px; right: 0px;\"></div>\n  <div class=\"slider-handle slider-handle-0 \"\n    tabindex=\"0\"\n    role=\"slider\"\n    aria-orientation=\"horizontal\"\n    aria-valuenow=\"10\"\n    aria-valuemin=\"0\"\n    aria-valuemax=\"20\"\n    style=\"position: absolute; z-index: 1; left: 385.5px;\">\n    <div class=\"tooltip top\">\n      <div class=\"tooltip-arrow\"></div>\n      <div class=\"tooltip-inner\">10</div>\n    </div>\n  </div>\n</div>"
      },
      "Should render disabled state": {
        "1": "<div class=\"slider-container disabled\"\n  style=\"position: relative;\">\n  <div class=\"slider-track active\"\n    style=\"position: absolute; left: 0px; right: 385.5px;\"></div>\n  <div class=\"slider-track\"\n    style=\"position: absolute; left: 385.5px; right: 0px;\"></div>\n  <div class=\"slider-handle slider-handle-0 \"\n    tabindex=\"0\"\n    role=\"slider\"\n    aria-orientation=\"horizontal\"\n    aria-valuenow=\"10\"\n    aria-valuemin=\"0\"\n    aria-valuemax=\"20\"\n    style=\"position: absolute; z-index: 1; left: 385.5px;\">\n    <div class=\"tooltip top\">\n      <div class=\"tooltip-arrow\"></div>\n      <div class=\"tooltip-inner\">10</div>\n    </div>\n  </div>\n</div>"
      },
      "Should render if set value": {
        "1": "<div class=\"slider-container\"\n  style=\"position: relative;\">\n  <div class=\"slider-track active\"\n    style=\"position: absolute; left: 0px; right: 693.9px;\"></div>\n  <div class=\"slider-track active\"\n    style=\"position: absolute; left: 77.1px; right: 385.5px;\"></div>\n  <div class=\"slider-track active\"\n    style=\"position: absolute; left: 385.5px; right: 192.75px;\"></div>\n  <div class=\"slider-track\"\n    style=\"position: absolute; left: 578.25px; right: 0px;\"></div>\n  <div class=\"slider-handle slider-handle-0 \"\n    tabindex=\"0\"\n    role=\"slider\"\n    aria-orientation=\"horizontal\"\n    aria-valuenow=\"2\"\n    aria-valuemin=\"0\"\n    aria-valuemax=\"20\"\n    style=\"position: absolute; z-index: 1; left: 77.1px;\">\n    <div class=\"tooltip top\">\n      <div class=\"tooltip-arrow\"></div>\n      <div class=\"tooltip-inner\">2</div>\n    </div>\n  </div>\n  <div class=\"slider-handle slider-handle-1 \"\n    tabindex=\"0\"\n    role=\"slider\"\n    aria-orientation=\"horizontal\"\n    aria-valuenow=\"10\"\n    aria-valuemin=\"0\"\n    aria-valuemax=\"20\"\n    style=\"position: absolute; z-index: 2; left: 385.5px;\">\n    <div class=\"tooltip top\">\n      <div class=\"tooltip-arrow\"></div>\n      <div class=\"tooltip-inner\">10</div>\n    </div>\n  </div>\n  <div class=\"slider-handle slider-handle-2 \"\n    tabindex=\"0\"\n    role=\"slider\"\n    aria-orientation=\"horizontal\"\n    aria-valuenow=\"15\"\n    aria-valuemin=\"0\"\n    aria-valuemax=\"20\"\n    style=\"position: absolute; z-index: 3; left: 578.25px;\">\n    <div class=\"tooltip top\">\n      <div class=\"tooltip-arrow\"></div>\n      <div class=\"tooltip-inner\">15</div>\n    </div>\n  </div>\n</div>"
      }
    }
  },
  "FileList": {
    "Display": {
      "Should render component filelist": {
        "1": "<div class=\"file-list table\">\n  <table>\n    <colgroup>\n      <col style=\"width: 4.4rem;\">\n      <col style=\"width: 6.2rem;\">\n      <col>\n      <col style=\"width: 16rem;\">\n      <col style=\"width: 5rem;\">\n      <col style=\"width: 5rem;\">\n    </colgroup>\n    <tbody>\n      <tr>\n        <td>1</td>\n        <td><i class=\"file-list-icon-1-c\"></i></td>\n        <td>test.dt<div class=\"file-list-subtitle\">Размер файла 123 мб</div>\n        </td>\n        <td>\n          <div class=\"file-list-tag\">Успешно<div class=\"file-list-subtitle\">\n              Тестовое пояснение к статусу</div>\n          </div>\n        </td>\n        <td class=\"with-button\"><button type=\"button\"\n            class=\"button-wrapper blank more\"><i\n              class=\"file-list-download\"></i></button></td>\n        <td class=\"with-button\"><button type=\"button\"\n            class=\"button-wrapper blank more\"><i\n              class=\"file-list-delete\"></i></button></td>\n      </tr>\n      <tr>\n        <td>2</td>\n        <td><i class=\"file-list-icon\"></i></td>\n        <td>test.test<div class=\"file-list-subtitle\">Размер файла 444 мб</div>\n        </td>\n        <td>\n          <div class=\"file-list-tag\">Ошибка<div class=\"file-list-subtitle\">\n              Тестовое пояснение к статусу</div>\n          </div>\n        </td>\n        <td class=\"with-button\"></td>\n        <td class=\"with-button\"><button type=\"button\"\n            class=\"button-wrapper blank more\"><i\n              class=\"file-list-delete\"></i></button></td>\n      </tr>\n      <tr>\n        <td>3</td>\n        <td><i class=\"file-list-icon-csv\"></i></td>\n        <td>test.csv<div class=\"file-list-subtitle\">Размер файла 666 мб</div>\n        </td>\n        <td>\n          <div class=\"file-list-tag success\">Успешно<div\n              class=\"file-list-subtitle\">Тестовое пояснение к статусу</div>\n          </div>\n        </td>\n        <td class=\"with-button\"></td>\n        <td class=\"with-button\"><button type=\"button\"\n            class=\"button-wrapper blank more\"><i\n              class=\"file-list-delete\"></i></button></td>\n      </tr>\n    </tbody>\n  </table>\n</div>"
      }
    }
  },
  "Password": {
    "Display": {
      "Should render the component": {
        "1": "<div class=\"password-element-wrapper required\"><input id=\"Password\"\n    data-test=\"password\"\n    placeholder=\"Enter your password...\"\n    aria-invalid=\"false\"\n    aria-required=\"true\"\n    class=\"password-input\"\n    form=\"AwesomePassword\"\n    name=\"Password\"\n    type=\"password\"\n    value=\"\"><i class=\"password-is-hidden\"></i></div>"
      },
      "Should render component with default value": {
        "1": "<div class=\"password-element-wrapper\"><input id=\"withDefaultValue\"\n    aria-invalid=\"false\"\n    class=\"password-input\"\n    type=\"password\"\n    value=\"Самый безопасный пароль\"><i class=\"password-is-hidden\"></i></div>"
      },
      "Should render clear button": {
        "1": "<div class=\"password-element-wrapper focused\"><input id=\"lowercase\"\n    aria-invalid=\"false\"\n    class=\"password-input password-input-clearable\"\n    type=\"password\"\n    value=\"hello\"><i class=\"password-clear-icon\"></i><i\n    class=\"password-is-hidden\"></i></div>"
      }
    }
  }
}
